src/Schema/SqliteSchemaManager.php

```diff
@@ -644,7 +646,7 @@ private function getForeignKeyDetails($table)
 
         for ($i = 0, $count = count($match[0]); $i < $count; $i++) {
             $details[] = [
-                'constraint_name' => isset($names[$i]) && $names[$i] !== '' ? $names[$i] : $i,
+                'constraint_name' => $names[$i] ?? '',
```

Comment:
probably fixed by https://github.com/doctrine/dbal/pull/5577

but `null` is used there, thus name may be regenerated

TODO for myself to rebase and retest